ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/ImmDB.hs

```diff
@@ -1,655 +0,0 @@
-              | realPointHash endPt == hash
-              -> ImmDB.iteratorClose it
-              | otherwise
-              -> return ()
-
--- | Stream headers/blocks after the given point
---
--- Returns 'ImmDB.WrongBoundError' if the lower bound is not part of the
--- ImmutableDB.
-streamAfter
-  :: forall m blk b.
-     (IOLike m, HasHeader blk, ImmDbSerialiseConstraints blk, HasCallStack)
-  => ImmDB m blk
-  -> ResourceRegistry m
-  -> BlockComponent (ChainDB m blk) b
-  -> Point blk -- ^ Exclusive lower bound
-  -> m (Either
-          (ImmDB.WrongBoundError (HeaderHash blk))
-          (ImmDB.Iterator (HeaderHash blk) m b))
-streamAfter db registry blockComponent low =
-    openIterator db registry blockComponent low' Nothing >>= \case
-      Left  err -> return $ Left err
-      Right itr -> do
-        case low of
-          GenesisPoint           -> return ()
-          -- Skip the exclusive lower bound
-          BlockPoint _slot _hash -> iteratorNext db itr >>= \case
-              ImmDB.IteratorExhausted ->
-                throwM $ ImmDbUnexpectedIteratorExhausted low
-              -- The hash must match the lower bound's hash. The ImmutableDB
-              -- already checks this.
-              ImmDB.IteratorResult {} -> return ()
-        return $ Right itr
-  where
-    low' :: Maybe (SlotNo, HeaderHash blk)
-    low' = case low of
-      GenesisPoint         -> Nothing
-      BlockPoint slot hash -> Just (slot, hash)
-
--- | Variation on 'streamAfter' that has the presence of the lower bound
--- as a precondition
---
--- Throws an exception if the precondition is violated.
-streamAfterKnownBlock
-  :: forall m blk b.
-     (IOLike m, HasHeader blk, ImmDbSerialiseConstraints blk, HasCallStack)
-  => ImmDB m blk
-  -> ResourceRegistry m
-  -> BlockComponent (ChainDB m blk) b
-  -> Point blk -- ^ Exclusive lower bound
-  -> m (ImmDB.Iterator (HeaderHash blk) m b)
-streamAfterKnownBlock db registry blockComponent low = do
-    mItr <- streamAfter db registry blockComponent low
-    case mItr of
-      Left  err -> throwM $ ImmDbMissingBlockPoint low err callStack
-      Right itr -> return itr
-
-{-------------------------------------------------------------------------------
-  Error handling
--------------------------------------------------------------------------------}
-
--- | Wrap calls to the ImmutableDB and rethrow exceptions that may indicate
--- disk failure and should therefore trigger recovery
-withDB :: forall m blk x. MonadCatch m
-       => ImmDB m blk -> (ImmutableDB (HeaderHash blk) m -> m x) -> m x
-withDB ImmDB{..} k = catch (k immDB) rethrow
-  where
-    rethrow :: ImmDB.ImmutableDBError -> m x
-    rethrow e = case wrap e of
-                    Just e' -> throwM e'
-                    Nothing -> throwM e
-
-    wrap :: ImmDB.ImmutableDBError -> Maybe ChainDbFailure
-    wrap (ImmDB.UnexpectedError e) = Just (ImmDbFailure e)
-    wrap ImmDB.UserError{}         = Nothing
```

Comment:
We no longer do this wrapping (not for the VolatileDB either). I should check what the impact of this is. The error policy has been updated accordingly, but I should check what happens when should an `ImmutableDBError` is thrown: is the ChainDB correctly closed?